### ISSUE TITLE
Mono Type.MakeArrayType allows arrays of byref types

### ISSUE THREAD

**Issue by hughbe** [dotnet/runtime#39001]
### Description
In https://github.com/dotnet/runtime/pull/36604
```cs
public static IEnumerable<object[]> MakeArrayType_ByRef_TestData()
{
    yield return new object[] { typeof(int).MakeByRefType() };
    yield return new object[] { typeof(TypedReference) };
    yield return new object[] { typeof(ArgIterator) };
    yield return new object[] { typeof(RuntimeArgumentHandle) };
    yield return new object[] { typeof(Span<int>) };
}

[Theory]
[MemberData(nameof(MakeArrayType_ByRef_TestData))]
public void MakeArrayType_ByRef_ThrowsTypeLoadException(Type t)
{
    Assert.Throws<TypeLoadException>(() => t.MakeArrayType());
}
```

```
System.Tests.TypeTests.MakeArrayType_ByRef_ThrowsTypeLoadException(t: typeof(System.Int32&)) [FAIL]
Assert.Throws() Failure Expected: typeof(System.TypeLoadException)
Actual:   (No exception was thrown)
Stack Trace:
/_/src/libraries/System.Runtime/tests/System/Type/TypeTests.cs(326,0): at
System.Tests.TypeTests.MakeArrayType_ByRef_ThrowsTypeLoadException(Type t) 
```

Should throw `TypeLoadException` as in .NET Core

**Comment by SamMonoRT** [comment]
@lambdageek - don't believe we really need this for 6.0, is my assumption right ?

**Comment by lambdageek** [comment]
Should be fixed by #55403
